base/sparse/sparsematrix.jl

```diff
@@ -2250,8 +2250,9 @@ setindex!(A::SparseMatrixCSC, x, i::Union{Integer, AbstractVector}, ::Colon) = s
 function setindex!{Tv}(A::SparseMatrixCSC{Tv}, x::Number,
         I::AbstractVector{<:Integer}, J::AbstractVector{<:Integer})
     if isempty(I) || isempty(J); return A; end
-    if !issorted(I); I = sort(I); end
-    if !issorted(J); J = sort(J); end
+    # lt=≤ to check for strict sorting
+    if !issorted(I, lt=≤); I = !allunique(I) ? sort!(unique(I)) : sort(I); end
```

Comment:
Thinking about this a bit more. The check for uniqueness is already allocating a `Set` and in any case you allocate a new array, so simply

```julia
if !issorted(I, lt=≤); I = sort!(unique(I)); end
```

should be good enough in general.